### PR TITLE
ENTESB-10724 - adds Publish and delete to virtualization detail

### PR DESCRIPTION
**Pull request by tejones**
Add Publish button and Delete menu option to virtualization detail page. Accessible from sql and view tabs.

Button and menu option are part of a new breadcrumb component for virtualization detail.

### CODE REVIEW

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.integration-detail__breadcrumb {
```
Comment (riccardo-forina) [inline app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.css]: Doesn’t this bleed in the integration editor? Why is it needed? Background colors should be set through a parent PageSection.

Also the Breadcrumb component will do same magic and inject the component in the root Page component, where the bg color is already set.

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.tsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  i18nUnpublish: string;
+  i18nUnpublishModalMessage: string;
+  i18nUnpublishModalTitle: string;
+  // icon?: string;
+  onDelete: (virtualizationName: string) => void;
+  /* TD-636: Commented out for TP
+  onExport: (virtualizationName: string) => void; */
+  onPublish: (virtualizationName: string, hasViews: boolean) => void;
+  onUnpublish: (virtualizationName: string) => void;
+  // publishingCurrentStep?: number;
```
Comment (mdrillin) [inline app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.tsx]: I'd prefer that the commented out properties be removed.  If we're sure they need to be added back later, then leave them but add a TODO

File: app/ui-react/syndesis/src/modules/data/pages/VirtualizationSqlClientPage.tsx [hunk 3]
```diff
@@ -40,65 +41,274 @@ export class VirtualizationSqlClientPage extends React.Component<
...
+                                      )}
+                                      i18nUnpublishModalTitle={t(
+                                        'virtualization.unpublishModalTitle'
+                                      )}
+                                      onDelete={handleDelete}
+                                      /* TD-636: Commented out for TP
+                                         onExport={this.handleExportVirtualization} */
+                                      onUnpublish={handleUnpublish}
+                                      onPublish={handlePublish}
+                                      // publishingLogUrl={
```
Comment (mdrillin) [inline app/ui-react/syndesis/src/modules/data/pages/VirtualizationSqlClientPage.tsx]: If not needed, please remove the commented out sections.

File: app/ui-react/syndesis/src/modules/data/pages/VirtualizationViewsPage.tsx [hunk 4]
```diff
@@ -97,253 +94,508 @@ export class VirtualizationViewsPage extends React.Component<
...
+                                                                     onExport={
+                                                                     this.handleExportVirtualization
+                                                                } */
+                                                                  onUnpublish={
+                                                                    handleUnpublish
+                                                                  }
+                                                                  onPublish={
+                                                                    handlePublish
+                                                                  }
+                                                                  // publishingLogUrl={
```
Comment (mdrillin) [inline app/ui-react/syndesis/src/modules/data/pages/VirtualizationViewsPage.tsx]: same comment as above, remove commented out sections - unless for sure needed, the make them a TODO

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.css (hunk 1 above)
Comment (tejones) [reply app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.css]: Correct.. thanks for that. Removed.

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.tsx (hunk 2 above)
Comment (tejones) [reply app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.tsx]: Yeah.. agreed. Removed comments and added TODOs.

File: app/ui-react/syndesis/src/modules/data/pages/VirtualizationSqlClientPage.tsx (hunk 3 above)
Comment (tejones) [reply app/ui-react/syndesis/src/modules/data/pages/VirtualizationSqlClientPage.tsx]: Removed and added TODOs

File: app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.css (hunk 1 above)
Comment (tejones) [reply app/ui-react/packages/ui/src/Data/Virtualizations/Views/ViewHeaderBreadcrumb.css]: Logged https://issues.jboss.org/browse/TEIIDTOOLS-659 to cover moving code to functional components. I'll make sure this happens before other tabs are created. Thanks @riccardo-forina .

**Comment by zregvart** [comment]
@tejones @mdrillin Can you create a PR to backport this to 1.7.x?

**Comment by zregvart** [comment]
This was included in `1.7.x` when it was branched from `master`, no need for a backport.

**Pull request merged**